save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt

```diff
@@ -249,23 +249,28 @@ class VulnerabilityService(
     }
 
     /**
-     * @param vulnerabilityDtos dto list of vulnerability projects
+     * @param vulnerabilityProjectDtos dto list of vulnerability projects
+     * @return [VulnerabilityDto] with saved [vulnerabilityProjectDtos]
      */
     @Transactional
-    fun saveAllProjects(vulnerabilityDtos: List<VulnerabilityProjectDto>) {
-        vulnerabilityRepository.findByName(vulnerabilityDtos.first().vulnerabilityName)?.let { vulnerability ->
-            vulnerabilityProjectRepository.saveAll(
-                vulnerabilityDtos.map { dto ->
-                    VulnerabilityProject(
-                        name = dto.name,
-                        url = dto.url,
-                        versions = dto.versions,
-                        type = dto.type,
-                        vulnerability = vulnerability,
-                    )
-                }
-            )
-        }
+    fun saveAllProjects(
+        vulnerabilityProjectDtos: List<VulnerabilityProjectDto>
+    ): VulnerabilityDto? = vulnerabilityRepository.findByName(vulnerabilityProjectDtos.first().vulnerabilityName)?.let { vulnerability ->
+        vulnerabilityProjectRepository.saveAll(
+            vulnerabilityProjectDtos.map { dto ->
+                VulnerabilityProject(
+                    name = dto.name,
+                    url = dto.url,
+                    versions = dto.versions,
+                    type = dto.type,
+                    vulnerability = vulnerability,
+                )
+            }
+        )
+            .map { it.toDto() }
+            .let { vulnerabilityProjects ->
+                vulnerability.toDtoWithDescription().copy(projects = vulnerabilityProjects)
```

Comment:
I think it is a bad idea.
The SaveAll method returns the saved list back to you. I think it should be transferred to the controller and from there, from the first project, get the vulnerability and already transfer it in the response to the front

Reply:
_@Transactional
    fun saveAllProjects(vulnerabilityDtos: List<VulnerabilityProjectDto>): List<VulnerabilityProject> {
        vulnerabilityRepository.findByName(vulnerabilityDtos.first().vulnerabilityName)?.let { vulnerability ->
            return  vulnerabilityProjectRepository.saveAll(...)
}_

Reply:
Will fix it

Reply:
Fixed as you suggested